src/main/java/picard/util/LiftoverUtils.java

```diff
@@ -390,20 +390,29 @@ protected static void leftAlignVariant(final VariantContextBuilder builder, fina
                     .map(a -> a.length)
                     .anyMatch(l -> l == 0)) {
                 // 6. extend alleles 1 nucleotide to the left
+
+
+                final byte extraBase;
+                final boolean extendLeft;
+                if (theStart > 1) {
+                    extraBase = referenceSequence.getBases()[theStart - 2];
+                    extendLeft = true;
+                    theStart--;
+                } else {
+                    extraBase = referenceSequence.getBases()[theEnd];
+                    extendLeft = false;
+                    theEnd++;
+                }
+
                 for (final Allele allele : alleleBasesMap.keySet()) {
                     // the first -1 for zero-base (getBases) versus 1-based (variant position)
                     // another   -1 to get the base prior to the location of the start of the allele
-                    final byte extraBase = (theStart > 1) ?
-                            referenceSequence.getBases()[theStart - 2] :
-                            referenceSequence.getBases()[theEnd];
-
-                    alleleBasesMap.put(allele, extendOneBase(alleleBasesMap.get(allele), extraBase));
+                    alleleBasesMap.put(allele, extendOneBase(alleleBasesMap.get(allele), extraBase, extendLeft));
                 }
-                changesInAlleles = true;
-                theStart--;
 
                 // 7. end if
             }
+            changesInAlleles = theStart!=oldStart || theEnd != oldEnd;
```

Comment:
This means repeat the above block of truncating matching right bases, and handling whatever empty alleles mean, whenever there's been a change in the variant. Is a `while` block the best way to handle this? Why not just truncate all matching bases on the right in one go, and then handle this other empty edge case as well?